From f8215757d52d41fe45aeb1eb21a4b3dc8f6f8314 Mon Sep 17 00:00:00 2001
From: John Crispin <john@phrozen.org>
Date: Mon, 9 Nov 2020 07:41:56 +0100
Subject: [PATCH] bluetooth: add tweak required for TIP

These are pending patches for OpenWrt HEAD to make bluetooth work.

Signed-off-by: John Crispin <john@phrozen.org>
---
 package/kernel/linux/modules/other.mk         | 14 +++-
 .../generic/hack-5.4/280-rfkill-stubs.patch   | 84 -------------------
 2 files changed, 10 insertions(+), 88 deletions(-)
 delete mode 100644 target/linux/generic/hack-5.4/280-rfkill-stubs.patch

diff --git a/package/kernel/linux/modules/other.mk b/package/kernel/linux/modules/other.mk
index c6952c812f..226895194b 100644
--- a/package/kernel/linux/modules/other.mk
+++ b/package/kernel/linux/modules/other.mk
@@ -14,10 +14,16 @@ define KernelPackage/6lowpan
   SUBMENU:=$(OTHER_MENU)
   TITLE:=6LoWPAN shared code
   KCONFIG:= \
-	CONFIG_6LOWPAN \
-	CONFIG_6LOWPAN_NHC=n
-  FILES:=$(LINUX_DIR)/net/6lowpan/6lowpan.ko
-  AUTOLOAD:=$(call AutoProbe,6lowpan)
+	CONFIG_6LOWPAN
+  FILES:=$(LINUX_DIR)/net/6lowpan/6lowpan.ko \
+	$(LINUX_DIR)/net/6lowpan/nhc_dest.ko \
+	$(LINUX_DIR)/net/6lowpan/nhc_fragment.ko \
+	$(LINUX_DIR)/net/6lowpan/nhc_hop.ko \
+	$(LINUX_DIR)/net/6lowpan/nhc_ipv6.ko \
+	$(LINUX_DIR)/net/6lowpan/nhc_mobility.ko \
+	$(LINUX_DIR)/net/6lowpan/nhc_routing.ko \
+	$(LINUX_DIR)/net/6lowpan/nhc_udp.ko
+AUTOLOAD:=$(call AutoProbe,6lowpan nhc_dest nhc_fragment nhc_hop nhc_ipv6 nhc_mobility nhc_routing nhc_udp)
 endef
 
 define KernelPackage/6lowpan/description
diff --git a/target/linux/generic/hack-5.4/280-rfkill-stubs.patch b/target/linux/generic/hack-5.4/280-rfkill-stubs.patch
deleted file mode 100644
index 2e48aea1cf..0000000000
--- a/target/linux/generic/hack-5.4/280-rfkill-stubs.patch
+++ /dev/null
@@ -1,84 +0,0 @@
-From 236c1acdfef5958010ac9814a9872e0a46fd78ee Mon Sep 17 00:00:00 2001
-From: John Crispin <john@phrozen.org>
-Date: Fri, 7 Jul 2017 17:13:44 +0200
-Subject: rfkill: add fake rfkill support
-
-allow building of modules depending on RFKILL even if RFKILL is not enabled.
-
-Signed-off-by: John Crispin <john@phrozen.org>
----
- include/linux/rfkill.h |  2 +-
- net/Makefile           |  2 +-
- net/rfkill/Kconfig     | 14 +++++++++-----
- net/rfkill/Makefile    |  2 +-
- 4 files changed, 12 insertions(+), 8 deletions(-)
-
---- a/include/linux/rfkill.h
-+++ b/include/linux/rfkill.h
-@@ -64,7 +64,7 @@ struct rfkill_ops {
- 	int	(*set_block)(void *data, bool blocked);
- };
- 
--#if defined(CONFIG_RFKILL) || defined(CONFIG_RFKILL_MODULE)
-+#if defined(CONFIG_RFKILL_FULL) || defined(CONFIG_RFKILL_FULL_MODULE)
- /**
-  * rfkill_alloc - Allocate rfkill structure
-  * @name: name of the struct -- the string is not copied internally
---- a/net/Makefile
-+++ b/net/Makefile
-@@ -53,7 +53,7 @@ obj-$(CONFIG_TIPC)		+= tipc/
- obj-$(CONFIG_NETLABEL)		+= netlabel/
- obj-$(CONFIG_IUCV)		+= iucv/
- obj-$(CONFIG_SMC)		+= smc/
--obj-$(CONFIG_RFKILL)		+= rfkill/
-+obj-$(CONFIG_RFKILL_FULL)	+= rfkill/
- obj-$(CONFIG_NET_9P)		+= 9p/
- obj-$(CONFIG_CAIF)		+= caif/
- ifneq ($(CONFIG_DCB),)
---- a/net/rfkill/Kconfig
-+++ b/net/rfkill/Kconfig
-@@ -2,7 +2,11 @@
- #
- # RF switch subsystem configuration
- #
--menuconfig RFKILL
-+config RFKILL
-+	bool
-+	default y
-+
-+menuconfig RFKILL_FULL
- 	tristate "RF switch subsystem support"
- 	help
- 	  Say Y here if you want to have control over RF switches
-@@ -14,19 +18,19 @@ menuconfig RFKILL
- # LED trigger support
- config RFKILL_LEDS
- 	bool
--	depends on RFKILL
-+	depends on RFKILL_FULL
- 	depends on LEDS_TRIGGERS = y || RFKILL = LEDS_TRIGGERS
- 	default y
- 
- config RFKILL_INPUT
- 	bool "RF switch input support" if EXPERT
--	depends on RFKILL
-+	depends on RFKILL_FULL
- 	depends on INPUT = y || RFKILL = INPUT
- 	default y if !EXPERT
- 
- config RFKILL_GPIO
- 	tristate "GPIO RFKILL driver"
--	depends on RFKILL
-+	depends on RFKILL_FULL
- 	depends on GPIOLIB || COMPILE_TEST
- 	default n
- 	help
---- a/net/rfkill/Makefile
-+++ b/net/rfkill/Makefile
-@@ -5,5 +5,5 @@
- 
- rfkill-y			+= core.o
- rfkill-$(CONFIG_RFKILL_INPUT)	+= input.o
--obj-$(CONFIG_RFKILL)		+= rfkill.o
-+obj-$(CONFIG_RFKILL_FULL)	+= rfkill.o
- obj-$(CONFIG_RFKILL_GPIO)	+= rfkill-gpio.o
-- 
2.25.1

